module/Frontpage/src/Frontpage/Controller/FrontpageController.php

```diff
@@ -3,13 +3,17 @@
 namespace Frontpage\Controller;
 
 use Zend\Mvc\Controller\AbstractActionController;
+use Zend\Session\Container as SessionContainer;
 use Zend\View\Model\ViewModel;
 
 class FrontpageController extends AbstractActionController
 {
     public function homeAction()
     {
-        return new ViewModel($this->getFrontpageService()->getHomepageData());
+        $homePageData = $this->getFrontpageService()->getHomepageData();
+        $session = new SessionContainer('lang');
+        $homePageData['lang'] = $session->lang;
+        return new ViewModel($homePageData);
```

Comment:
You can obtain the language from the translator / locale in the view directly. Using this code:

```php
$this->plugin('translate')->getTranslator()->getLocale();
```